code/obj/item/swingsign.dm

```diff
@@ -0,0 +1,190 @@
+				return
+			random_brute_damage(M, rand(1,maxcrashdamage),1)
+			L.force_laydown_standup()
+			L.changeStatus("stunned", 1 SECONDS)
+			src.visible_message("<b><font color=red>[M] is people-stopped by [src]!</font></b>")
+			playsound(src, 'sound/impact_sounds/Wood_Hit_1.ogg',50,1)
+			fold() //Change to to item
+			return
+		..()
+
+	/obj/swingsign/custom_suicide = 1
+	///obj/swingsign/
+	suicide(var/mob/user as mob)
+		if (!src.user_can_suicide(user))
+			return 0
+		user.visible_message("<span class='alert'><b>[user] puts [his_or_her(user)] head between [src]'s legs and clamps them shut!</b></span>")
+		user.TakeDamage("head", 250, 0)
+		playsound(src, 'sound/items/woodbat.ogg',50,1)
+		fold()
+		return 1
+
+	//UI =============================
+
+	//Lord give me patience for this bs
+	/obj/swingsign/ui_interact(mob/user, datum/tgui/ui)
+		ui = tgui_process.try_update_ui(user, src, ui)
+		if(!ui)
+			ui = new(user, src, "SwingSign")
+			ui.open()
+
+	/obj/swingsign/ui_data(mob/user)
+		. = list(
+			"message" = src.message,
+			"maxRows" = src.maxmessagerows,
+			"maxCols" = src.maxmessagecols
+		)
+
+	/obj/swingsign/ui_act(action, params)
+		. = ..()
+		if (.)
+			return
+		if(action == "save_message")
+			var/new_message = params["message"]
+			//if(length(new_message) > maxmessagelength) //Length checks moved to SwingSigh.js
+			//	return FALSE
+			setmessage(new_message)
+			. = TRUE
+		update_icon()
+
+	//OTHER PROCS ====================
+	proc/setmessage(var/newmessage)
+		message = newmessage
+		if(message == "")
+			desc = defaultdesc
+			icon_state = "blank"
+		else
+			desc = descpreamble + message + "</div>"
+			icon_state = "written"
+
+	proc/fold()
+		var/obj/item/swingsignfolded/C = new/obj/item/swingsignfolded(src.loc)
+
+		if (src.material)
+			C.setMaterial(src.material)
+		if (src.icon_state)
+			C.icon_state = "folded"
+			//C.item_state = C.icon_state
+		if (src.message)
+			C.message = src.message
+
+		qdel(src)
+
+//Heldable folded sign ==============
+TYPEINFO(/obj/item/swingsignfolded)
+	mat_appearances_to_ignore = list("wood")
+/obj/item/swingsignfolded
+	name = "swing sign"
+	desc = "A foldable sign for writing annoucements and advertisements"
+	icon = 'icons/obj/furniture/swingsign.dmi'
+	icon_state = "folded"
+	inhand_image_icon = 'icons/obj/furniture/swingsign.dmi'
+	item_state = "folded_held"
+	w_class = W_CLASS_BULKY
+	throwforce = 10
+	flags = FPRINT | TABLEPASS
+	force = 5
+	stamina_damage = 45
+	stamina_cost = 21
+	stamina_crit_chance = 10
+	material_amt = 0.1
+	hitsound = 'sound/impact_sounds/folding_chair.ogg'
+
+	var/message = ""//Stored message for deployed object
+
+	New()
+		..()
+		src.setItemSpecial(/datum/item_special/swipe)
+		BLOCK_SETUP(BLOCK_LARGE)
+
+	/obj/item/swingsignfolded/attack_self(mob/user as mob)
+		if(cant_drop == 1)
+			boutput(user, "You can't unfold the [src] when its attached to your arm!")
+			return
+		var/obj/swingsign/S = null
+		S = new/obj/swingsign/(user.loc)
```

Comment:
```suggestion
		var/obj/swingsign/sign = null
		sign = new(user.loc)
```
Don't use single letter variable names except for standard ones like `T` for turf, `M` for mob etc.